assembly/src/assembler/mod.rs

```diff
@@ -309,6 +419,20 @@ impl Assembler {
         ))
     }
 
+    /// Compile all of the uncompiled procedures in the module graph, placing them
+    /// in the procedure cache once compiled.
+    ///
+    /// Returns an error if any of the provided Miden Assembly is invalid.
+    fn assemble_graph(
+        &mut self,
+        mast_forest_builder: &mut MastForestBuilder,
+    ) -> Result<(), Report> {
+        let mut worklist = self.module_graph.topological_sort().to_vec();
+        assert!(!worklist.is_empty());
+        self.process_graph_worklist(&mut worklist, None, mast_forest_builder)
+            .map(|_| ())
+    }
```

Comment:
I think the reference to the procedure cache in the comment needs to be updated.

Also, question: the difference between this method and `compile_subgraph()` is that this one will output MAST for all modules in the module graph, but `compile_subgraph()` outputs only things related to the `entrypoint`? If so, is `is_entrypoint` parameter there somewhat redundant?

Reply:
`compile_subgraph` is used to compile individual procedures, whereas `compile_graph` does the entire module graph.

Previously, `assemble_program` invoked `compile_subgraph` on the entrypoint, so as to only include things reachable from the entrypoint in the generated `Program`. Similarly, `assemble_kernel` invoked `compile_subgraph` on each of the kernel module exports, so as to only include things reachable from those exports.

Essentially, `compile_subgraph` allows us to be precise about what ends up in the compiled output; while `compile_graph` assumes we want everything in the module graph to be available in the procedure cache. Now that the procedure cache isn't really a thing, the need for `compile_graph` is mostly gone, and `compile_subgraph` is what we want to use (on the specific procedures which we want to include in the compiled output).

Reply:
Removed `assemble_graph()`, and this cleaned things up quite a bit (e.g. we no longer need `Assembler::find()` or `Assembler::get_module_exports()`. Updated `assemble_library()` accordingly.

This also meant that `ModuleGraph.topo` was no longer used, so I removed it too.